src/library/scala/collection/Map.scala

```diff
@@ -207,6 +211,16 @@ trait MapOps[K, +V, +CC[_, _] <: IterableOps[_, AnyConstr, _], +C]
   }
 
   /** An [[Iterable]] collection of the keys contained by this map.
+   *
+   *  The default implementation returns `keySet`.
+   *
+   *  For `Map` implementations with ordered but not sorted keys, such as a `SeqMap`,
+   *  `keys` may efficiently build a representation that preserves order.
+   *
+   *  Its iterator respects the iteration order of `keysIterator`, such that
+   *  ```
+   *  keys.iterator.sameElements(keySet.iterator)
+   *  ```
```

Comment:
Is that a property that always holds? "Its iterator" - what is that referring to?

Reply:
The subject is `keys`, hence `keys.iterator`. This is the property, yes. That is, don't use `equals` as in OP.